server/src/models/budget.js

```diff
@@ -1,31 +1,24 @@
 const fs = require('fs');
 const path = require('path');
+const dbConnection = require('../database/db_connection');
 const BUDGET_PATH = path.resolve('./src/budget.json');
 const dataCopy = require(BUDGET_PATH);
-const {
-  insertToBudget
-} = require("../queries/postData");
 
-exports.getBudgetData = () => {
-  const data = require(BUDGET_PATH);
-  return data;
+exports.getBudgetData = cb => {
+  dbConnection.query('SELECT * FROM budget', (err, result) => {
+    if (err) return cb(err);
+    return cb(null, result.rows);
+  });
 };
 
-
- 
-
 exports.addBudgetItem = async (itemName, quantity, price, category) => {
-  //I added the call to the DB insert query inside the function that inserts data to Json file, 
-  //so it is called once the API fitched from frontend
-  //send userId hardcoded until we send it with the cookie
-  const result = insertToBudget(1,itemName, quantity, price, category,(err, res) => {
+  const result = insertToBudget(1, itemName, quantity, price, category, (err, res) => {
     if (err) {
       return err;
     }
     return res;
   });
- 
-  
+
   return new Promise((resolve, reject) => {
```

Comment:
your function ALWAYS retuns line 19 or line or line 17
everything aftrer line 20 is useless in this case